### ISSUE TITLE
Bug: Argument type CompatibleConfigArray is not assignable to parameter type InfiniteDepthConfigWithExtends

### ISSUE THREAD

**Issue by rodrigoreis** [typescript-eslint/typescript-eslint#11519]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=FBA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AQ2mncNqNPRREXAPbRIAGnDY8-clRr1kjFhwDu7SnXwALRE3YAjJIMjDoYyNIC%2BIa0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
Seems that change from `TSESLint.FlatConfig.ConfigArray` to `CompatibleConfigArray` default configs is now warning:

Argument type `CompatibleConfigArray` is not assignable to parameter type `InfiniteDepthConfigWithExtends`
```

### ESLint Config

```javascript
import tseslint from 'typescript-eslint';

export default _eslint.config(
    tseslint.configs.strictTypeChecked,
    tseslint.configs.stylisticTypeChecked
);
```

### tsconfig

```jsonc

```

### Expected Result

No warnings

### Actual Result

Argument type `CompatibleConfigArray` is not assignable to parameter type `InfiniteDepthConfigWithExtends`

**Comment by JoshuaKGoldberg** [comment]
> playground
> repro

@rodrigoreis that playground link is blank, and the ESLint config references an `_eslint` variable that isn't defined in what you pasted. We can't help you without being able to see the actual code you're working with. Could you please post a full reproduction? 

https://antfu.me/posts/why-reproductions-are-required

**Comment by JoshuaKGoldberg** [comment]
Closing out as it's been >1 month without response. If anybody has a reproduction, please do post it - we'd be happy to take a look. Cheers!
